man/tpm2_getekcertificate.1.md

```diff
@@ -0,0 +1,63 @@
+% tpm2_getekcertificate(1) tpm2-tools | General Commands Manual
+
+# NAME
+
+**tpm2_getekcertificate**(1) - Retrieve the Endorsement key Certificate for the TPM
+endorsement key from the TPM manufacturer's endorsement certificate hosting
+server.
+
+# SYNOPSIS
+
+**tpm2_getekcertificate** [*OPTIONS*] _EK\_PUBLIC\_FILE_
```

Comment:
I am not 100% sure what the _EK\_PUBLIC\_FILE_ is as it's not documented nor in the examples on how obtain it. Im assuming its the public portion of the object. Make the argument the URL and make the ek an argument, so it just follows the normal -u convention. If it's supposed to be ek context, then use -c.

Reply:
> > @williamcroberts Makefile.am changes to move the tool to misc in commit [dddd6c3](https://github.com/tpm2-software/tpm2-tools/commit/dddd6c39688bdb68ed4889b8837a8a023d604cfa) is causing the compiled tool to show in both tools and tools/misc folder. Any thoughts?
> 
> Probably didn't clean you tree before moving and recompiling so it's old. If you look at the timestamp is it the same as the other tool?

I did do a clean and distclean prior to compiling again. The timestamps match too. The only difference I can tell from the other misc/tools is that the getmanufec( or now getekcertificate) has CFLAGS and LDADD that I am also moving to tools_misc_tpm2_getekcertificate

Reply:
> I am not 100% sure what the _EK_PUBLIC_FILE_ is as it's not documented nor in the examples on how obtain it. Im assuming its the public portion of the object. Make the argument the URL and make the ek an argument, so it just follows the normal -u convention. If it's supposed to be ek context, then use -c.

Yes it exclusively requires 2 inputs -
1. The public portion of the endorsement key and 
2. The portal address that host the certificate for the public key.

I will update the example section to show how to retrieve the EK_PUBLIC_FILE. Should this then still follow the -u convention. I was thinking from a factory line perspective where you have one internet facing server to retrieve multiple ek certificates. If the factory used a script the only variable would be the ek public of the various platforms which can be passed as an argument.

Reply:
Updated example section

Reply:
flip so the URL is not an option, the pub is via -u